src/model/wallet.ts

```diff
@@ -300,11 +293,11 @@ export const loadWallet = async <S extends Screen>({
   if (isHardwareWalletKey(privateKey)) {
     const index = privateKey?.split('/')[1];
     const deviceId = privateKey?.split('/')[0];
-    if (typeof index !== undefined && provider && deviceId) {
+    if (typeof index !== undefined && deviceId && provider) {
       return new LedgerSigner(provider, getHdPath({ type: WalletLibraryType.ledger, index: Number(index) }), deviceId);
     }
   } else if (privateKey) {
-    return new Wallet(privateKey, provider || web3Provider);
```

Comment:
This is the line I mentioned before. Now the provider defaults to mainnet instead of being undefined.